feat: Create Bepu demo sample templates.

Doprez opened the pull request:
# PR Details

This PR re-configures the old demos into a project template for users to look at how some of the features are meant to work in a playable environment.

I have also updated Bepu to the latest version which seems to fix the convex creation issue.

## Related Issue

There have been a few mentions in discord about the lack of docs/examples for the Bepu library.
https://github.com/stride3d/stride/issues/2703

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My change requires a change to the documentation.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] **I have built and run the editor to try this change out.**

Doprez commented:
Couple more things to fix:
- Controls per scene dont work (Should be simple to fix I just made the main camera centralized which is the issue)
- Car scene crashes on load due to https://github.com/stride3d/stride/issues/2703

Doprez commented:
This is pretty much done but the Car scene still has the crash issue. It was happening before the changes here though so it is not an issue created in this project. (Unless there was something done incorrectly in the Car scene specifially of course)

Eideren commented:
Doesn't look like this is part of the samples that the project window can create, I might be missing something though ... ?

Doprez commented:
Sorry, you were not missing anything. I forgot to add the reference to the template file for the launcher.

Although I noticed with the template its improperly creating a separate Windows project so Im just looking into that now.

Doprez commented:
Ok so it turns out you **must** add the references to the non main scenes as root assets in the sdpkg for the template to work properly. 

One final issue is the resource files not being added for some reason. Almost there though.

Doprez commented:
ok, **now** this should be mostly working again apart form the car demo due to the previous issue. Sorry again Eideren, I should have made sure to properly test the template in the launcher not just the project.

Eideren commented:
All good, getting a 
```
[EXEC(0,0)]: Error: [ModelAssetCompiler] Unable to find the source file 'D:/Documents/Stride Projects/BepuSample/Resources/Models/Hex.obj' for Asset [[ModelAsset] Models/Hex]
```
now when building the project, same with the dungeon and hex models

Doprez commented:
hmmm 
![image](https://github.com/user-attachments/assets/ebd861ac-7889-4494-a7f4-c148369b33cb)

Also tested in my Document folder just in case:
![image](https://github.com/user-attachments/assets/e58fd1fc-93df-4463-9f69-f84d83873139)

Opened the scene with the models and it looks fine:
![image](https://github.com/user-attachments/assets/7ee7af5f-2500-4043-bc97-9e6707a72eae)

It also seems to load at runtime as well. Can you confirm if you see the assets in the projects resource folder? `{ProjectName}\Resources\Models`

Doprez commented:
good catch, for some reason the gitignores had .obj files? Not usre if there is some greater risk with removing those lines but it only added the ones I wanted in this case.

Eideren commented:
The samples are a bit too rough right now I think. The car scene, for example, captures the mouse on start, if the user presses tab the mouse is unlocked but the scene selection menu disappears at the same time so we can't easily go back to another scene.

I think we should remove the weird platform from the car scene as well given how much of a performance issue it is at the start but even afterward. Doesn't make for a good best practice example.

And we'll also have to wait for #2722 to be merged before merging this one.

Doprez commented:
Sounds good, I will clean it up with a proper state machine and maybe just merge the FPS and Car scene into one in that case, it may be more interesting for users anyway.

And thank you for figuring out that crash!

Doprez commented:
I cleaned up the demo and removed the character and car examples as I have a plan to make a separate Bepu template dedicated to being more of a simple game example.

I will make a new PR for a template that will have an first person controller that can enter and exit a car and depepdning on my time I will try to add the same shooter functionality from the FPS demo that exists with bullet.

Doprez commented:
Oh I also added a couple more constraint examples (ball socket and distance).

Kryptos-FR commented on .gitignore:
```diff
@@ -11,7 +11,6 @@ _ReSharper.*
 *.Cache
 *.tlog
 *.log
```
This shouldn't be removed. the `.obj` extension is found in C/C++ projects for instance.

You don't need to touch the .gitignore file even if you want to add a file which extension is ignored. It can be done from the command line (using the `-f` switch).

Doprez replied:
Should be back. Its good to know about the C++ stuff I figured it was just for projects without LFS set up.

Kryptos-FR commented on samples/.gitignore:
```diff
@@ -11,7 +11,6 @@ _ReSharper.*
 *.cache
 *.tlog
 *.log
-*.obj
```
Same remark. As we want this `.gitignore` file to be a sample by itself, we shouldn't touch it.

Eideren commented:
Thanks !
